apps/src/code-studio/ConfirmEmailModal.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import locale from '@cdo/locale';
+import color from '@cdo/apps/util/color';
+
+const styles = {
+  title: {
+    position: 'absolute',
+    left: 30,
+    top: 10,
+    color: color.dark_charcoal,
+    margin: '15px 0',
+  },
+  content: {
+    position: 'absolute',
+    left: 30,
+    top: 50,
+    right: 30,
+    bottom: 70
+  },
+  contentText: {
+    fontSize: 15,
+    lineHeight: '22px'
+  },
+  highlightRow: {
+    backgroundColor: color.default_blue,
+    color: color.white,
+  },
+  footer: {
+    position: 'absolute',
+    bottom: 25,
+    right: 30,
+    left: 30,
+  },
+  buttonPrimary: {
+    float: 'right',
+    background: color.orange,
+    color: color.white,
+    border: '1px solid #b07202',
+    borderRadius: 3,
+    boxShadow: 'inset 0 1px 0 0 rgba(255, 255, 255, 0.63)',
+    fontSize: 14,
+    padding: '8px 20px',
+  },
+  buttonSecondary: {
+    float: 'left',
+    background: '#eee',
+    color: '#5b6770',
+    border: '1px solid #c5c5c5',
+  },
+  hr: {
+    borderColor: color.light_gray
+  },
+  email: {
+    height: 35,
+    paddingLeft: 15
+  }
+};
+
+/**
+ * Pops up a dialog that prompts the user to confirm their email address.
+ * This is used when oauth accounts switch from student to teacher, in order
+ * to verify that the email address is already known to the user (it will
+ * become visible on the accounts page after the transition, which is a
+ * potential violation of student privacy).
+ */
+export default class ConfirmEmailModal extends React.Component {
+  static propTypes = {
+    handleSubmit: React.PropTypes.func,
+    handleCancel: React.PropTypes.func,
+    isOpen: React.PropTypes.bool,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  cancel = () => {
```

Comment:
Actually, I'm wrong - see http://egorsmirnov.me/2015/08/16/react-and-es6-part3.html

This pattern is used to get around autobinding differences.